app/controllers/socials.py

```diff
@@ -0,0 +1,120 @@
+        filter_type = request.args.get('filter', '').lower() or None
+        
+        page = request.args.get('page', 1, type=int)
+        per_page = request.args.get('per_page', 10, type=int)
+        
+        # Validate per_page limits
+        if per_page > 100:
+            per_page = 100
+        elif per_page < 1:
+            per_page = 10
+            
+        # Validate page
+        if page < 1:
+            page = 1
+        
+        # Validate entity parameter
+        valid_entities = ['projects', 'users', 'tags']
+        if entity and entity not in valid_entities:
+            return dict(
+                status="fail", 
+                message=f"Invalid entity. Must be one of: {', '.join(valid_entities)}"
+            ), 400
+        
+        # Validate filter parameter
+        valid_filters = ['trending', 'recently_updated', 'newly_added']
+        if filter_type and filter_type not in valid_filters:
+            return dict(
+                status="fail", 
+                message=f"Invalid filter. Must be one of: {', '.join(valid_filters)}"
+            ), 400
+        
+        try:
+            if entity:
+                if entity == 'projects':
+                    result = get_projects_data(current_user, search, filter_type, page, per_page)
+                elif entity == 'users':
+                    result = get_users_data(current_user, search, filter_type, page, per_page)
+                elif entity == 'tags':
+                    result = get_tags_data(current_user, search, filter_type, page, per_page)
+                
+                return dict(status='success', data=result), 200
+            else:
+                combined_per_page = per_page
+ 
+                projects_list = get_projects_data(
+                    current_user, search, filter_type, 
+                    page=page, per_page=combined_per_page, paginate=False
+                )
+                users_list = get_users_data(
+                    current_user, search, filter_type, 
+                    page=page, per_page=combined_per_page, paginate=False
+                )
+                tags_list = get_tags_data(
+                    current_user, search, filter_type, 
+                    page=page, per_page=combined_per_page, paginate=False
+                )
+                
+                # Ensure we have lists
+                if not isinstance(projects_list, list):
+                    projects_list = []
+                if not isinstance(users_list, list):
+                    users_list = []
+                if not isinstance(tags_list, list):
+                    tags_list = []
+  
+                # Calculate total counts for proper pagination
+                total_projects = len(projects_list)
+                total_users = len(users_list)
+                total_tags = len(tags_list)
+                total_items = total_projects + total_users + total_tags
+                
+                result = {
```

Comment:
You should return the page the user is on please, total is not the best way to total_items  as it could be confused for total record available